File: 20220222-java-virtual-storage-design-kata/src/test/java/com/katas/StorageTest.java
package com.katas;

import org.junit.Test;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class StorageTest {

    private Storage storage = new Storage();

    @Test
    public void should_have_a_root_directory() {
        Directory directory = storage.rootDirectory();

        assertThat(directory.name()).isEqualTo("/");
    }

    @Test
    public void should_have_a_total_size_of_zero_when_is_empty() {
        assertThat(storage.totalSize()).isZero();
    }

    @Test
    public void should_have_a_total_size_when_non_empty() {
        // given
        storage.rootDirectory().add(new File("file1", 100));

        Directory dir1 = new Directory("dir1");
        dir1.add(new File("file1", 50));
        storage.rootDirectory().add(dir1);

        // when
        int totalSize = storage.totalSize();

        // then
        assertThat(totalSize).isEqualTo(150);
    }
}

File: 20200807-kotlin-virtual-disk/src/test/kotlin/com/katas/StorageTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class StorageTest {

    private val storage = Storage()

    @Test
    fun `should have a root directory`() {
        assertThat(storage.rootDirectory()).isEqualTo(Directory("/"))
    }

    @Test
    fun `should calculate the total size used by its contents`() {
        storage.rootDirectory().add(File(name = "file1", size = 100))
        val directory = Directory("dir1")
        directory.add(File(name = "file2", size = 50))
        storage.rootDirectory().add(directory)

        assertThat(storage.totalSize()).isEqualTo(150)
    }

    @Test
    fun `should calculate the total size used when is empty`() {
        assertThat(storage.totalSize()).isZero()
    }

    @Test
    fun `should count the total number of mp3 files`() {
        storage.rootDirectory().add(File(name = "file1.mp3", size = 1))
        storage.rootDirectory().add(File(name = "file2.txt", size = 2))

        assertThat(storage.totalMP3()).isOne()
    }
}